go-controller/pkg/util/multi_network.go

```diff
@@ -957,3 +965,7 @@ func DoesNetworkRequireTunnelIDs(netInfo NetInfo) bool {
 	// Layer2Topology with IC require that we allocate tunnel IDs for each pod
 	return netInfo.TopologyType() == types.Layer2Topology && config.OVNKubernetesFeature.EnableInterconnect
 }
+
+func AllowsPersistentIPs(netInfo NetInfo) bool {
+	return netInfo.IsPrimaryNetwork() && netInfo.TopologyType() == types.Layer2Topology && netInfo.AllowsPersistentIPs()
```

Comment:
Shouldn't this cover secondary networks as well?

Reply:
I'd rather not; I would prefer to have an helper tailored for this use: primary layer2 UDN with persistent IPs. 

I can make the name more appropriate though.

Reply:
It doesn't make sense then because what I wanted help with was: ` summarizes which networks allow persistent IPs`

Reply:
So let's put it back the way it was originally, because it is different in primary UDNs and in secondary networks (in primary UDNs it is only for layer2; in secondary networks it allows for localnet networks as well).

My bad in blindly going forward with this.

Reply:
Done.

Reply:
> So let's put it back the way it was originally, because it is different in primary UDNs and in secondary networks (in primary UDNs it is only for layer2; in secondary networks it allows for localnet networks as well).
> 
> My bad in blindly going forward with this.

That's actually the nuance that I wanted to put on the table and summarize in a single place. Wouldn't this work, for example:
```
func AllowsPersistentIPs(netInfo NetInfo) bool {
  switch {
      case netInfo.IsPrimaryNetwork():
          return netInfo.TopologyType() == types.Layer2Topology && netInfo.AllowsPersistentIPs()
  
      case netInfo.IsSecondaryNetwork():
          return (netInfo.TopologyType() == types.Layer2Topology || netInfo.TopologyType() == types.LocalnetTopology) && netInfo.AllowsPersistentIPs()
  
      default:
          return false
  }
}
```

Reply:
OK, I'll do this.

Reply:
Done; the other usages are just too weirdly scoped for me to change in the context of this PR.

Reply:
what about cluster manager? Aren't layer 2 networks reconciling ipam claims from there? This might have changed and I missed it.

Reply:
Correct; done.